src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -480,7 +480,7 @@ protected function initializeBundles()
      */
     protected function getContainerClass()
     {
-        return $this->name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'ProjectContainer';
+        return $this->name.ucfirst(preg_replace('/[^a-zA-Z0-9_]/', '', $this->environment)).($this->debug ? 'Debug' : '').'ProjectContainer';
```

Comment:
the regex about valid chars in class names is broader than that

Reply:
The official regexp for class names is `^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$` (see http://php.net/manual/en/language.oop5.basic.php) so this part should be `[^a-zA-Z0-9_\x7f-\xff]` instead of `[^a-zA-Z0-9_]` ... but I don't think we should do this change. The probability of causing a clash because two weird env names have been stripped from wrong characters is negligible.